### PR TITLE
Split up fields into different field types for Painless Fields API

### PR DESCRIPTION
**Pull request by jdconrad**
This change refactors the inner fields classes and inner converter classes into field class split by type for the Painless Fields API.

Each field class now contains the following:
* it's own members
* a converter that converts from other types of fields to its own type of field
* conversion helper methods to go from its own type of field to other types of fields
* conversion helper methods to go from its own type to other types

### CODE REVIEW

File: modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt [hunk 1]
```diff
@@ -10,8 +10,8 @@
 
 # API
 class org.elasticsearch.script.field.Field {
-  org.elasticsearch.script.field.Converter BigInteger
-  org.elasticsearch.script.field.Converter Long
+  org.elasticsearch.script.field.Converter BigInteger @augmented[augmented_canonical_class_name="org.elasticsearch.script.field.BigIntegerField"]
```
Comment (stu-elastic) [inline modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt:13]: What's going on here?  I don't see BigInteger on Field.

Comment (jdconrad) [reply modules/lang-painless/src/main/resources/org/elasticsearch/painless/org.elasticsearch.script.fields.txt:13]: I moved these singletons to their respective field classes. The allowlist in Painless uses the augmented fields for this separation. Note this is consistent with any extension of field outside of core as well like UnsignedLongField.

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java [hunk 2]
```diff
@@ -9,9 +9,105 @@
...
+    static {
+        BigInteger = new Converter<BigInteger, BigIntegerField>() {
+            @Override
+            public BigIntegerField convert(Field<?> sourceField) {
+                if (sourceField instanceof LongField) {
+                    return LongField.convertLongToBigIntegerField((LongField) sourceField);
+                }
+
+                if (sourceField instanceof DoubleField) {
+                    return DoubleField.convertDoubleToBigIntegerField((DoubleField) sourceField);
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java:35]: How about `DoubleField.toBigIntegerField`?

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java [hunk 3]
```diff
@@ -9,9 +9,105 @@
...
+                    return StringField.convertStringToBigIntegerField((StringField) sourceField);
+                }
+
+                if (sourceField instanceof DateMillisField) {
+                    return LongField.convertLongToBigIntegerField(
+                            DateMillisField.convertDateMillisToLongField((DateMillisField) sourceField));
+                }
+
+                if (sourceField instanceof DateNanosField) {
+                    return LongField.convertLongToBigIntegerField(
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java:48]: What's the thinking here?

Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java:48]: I dropped this a line to fit into our character limits, but functionally didn't change anything other than the location of the method.

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java (hunk 2 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java:35]: I like that better too. I will make the change.

File: server/src/main/java/org/elasticsearch/script/field/EmptyField.java [hunk 4]
```diff
@@ -15,6 +15,9 @@
  * Script field with no mapping, always returns {@code defaultValue}.
  */
 public class EmptyField<T> extends Field<T> {
+
+    /* ---- Empty Field Members ---- */
```
Comment (stu-elastic) [inline server/src/main/java/org/elasticsearch/script/field/EmptyField.java:19]: What's this mean?

Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/EmptyField.java:19]: Ah, I added this for consistency with the other fields where I was marking sections with the actual field members, converter helper methods, and converter. Happy to remove this and any others if you want. For the others I find it easier to scan the file with these comments as markers and it creates a consistency in ordering through each field class.

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java (hunk 2 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java]: Pushed the change for this.

File: server/src/main/java/org/elasticsearch/script/field/EmptyField.java (hunk 4 above)
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/script/field/EmptyField.java:19]: Ya remove the comment where it doesn't make sense.

Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/EmptyField.java:19]: Removed.

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java (hunk 2 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java]: Changed the type conversion methods as well to this format.

File: server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java (hunk 3 above)
Comment (jdconrad) [reply server/src/main/java/org/elasticsearch/script/field/BigIntegerField.java]: Moved the line back up with the name changes, but did not change the logic.

**Review by stu-elastic — APPROVED** [review]
Nice!

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review! Will commit as soon as CI passes.

**Pull request merged**